src/main/java/net/minecraftforge/eventbus/api/EventListenerHelper.java

```diff
@@ -53,6 +53,12 @@ public static ListenerList getListenerList(Class<?> eventClass)
     static ListenerList getListenerListInternal(Class<?> eventClass, boolean fromInstanceCall)
     {
         if (eventClass == Event.class) return EVENTS_LIST; // Small optimization, bypasses all the locks/maps.
+
+        // Skip allocating lambda if possible
+        var list = listeners.get(eventClass);
+        if (list != null)
+            return list;
+
         return listeners.computeIfAbsent(eventClass, () -> computeListenerList(eventClass, fromInstanceCall));
```

Comment:
If `computeListenerList` never returns null, would replacing the `computeIfAbsent` with a simple `put` improve it slightly further?
```java
list = computeListenerList(eventClass, fromInstanceCall);
listeners.put(eventClass, list);
return list;
```

Though guessing that might not be as helpful? (Especially given a wrapper for `put` would have to be written for `LockHelper`)

Reply:
We could do `putIfAbsent` but it would have to be written in `LockHelper`.

`put` will not work correctly with multithreading - we need to ensure that there is a shared `ListenerList` instance.